cmd/admin-heal-start.go

```diff
@@ -0,0 +1,153 @@
+	Before:          setGlobalsFromContext,
+	Action:          mainAdminHealStart,
+	Flags:           append(adminHealStartFlags, globalFlags...),
+	HideHelpCommand: true,
+	CustomHelpTemplate: `NAME:
+  {{.HelpName}} - {{.Usage}}
+
+USAGE:
+  {{.HelpName}} [FLAGS] TARGET
+
+FLAGS:
+  {{range .VisibleFlags}}{{.}}
+  {{end}}
+EXAMPLES:
+    1. To format newly replaced disks in a Minio server with alias 'play'
+       $ {{.HelpName}} play
+
+    2. Heal 'testbucket' in a Minio server with alias 'play'
+       $ {{.HelpName}} play/testbucket/
+
+    3. Heal all objects under 'dir' prefix
+       $ {{.HelpName}} --recursive play/testbucket/dir/
+
+    4. Heal all uploads under 'dir' prefix (WIP)
+       $ {{.HelpName}} --incomplete --recursive play/testbucket/dir/
+
+    5. Issue a dry-run heal operation to list all objects to be healed
+       $ {{.HelpName}} --dry-run play
+
+    6. Issue a dry-run heal operation to list all uploads to be healed (WIP)
+       $ {{.HelpName}} --dry-run --incomplete play
+
+    7. Issue a dry-run heal operation to list all objects to be healed under 'dir' prefix
+       $ {{.HelpName}} --recursive --dry-run play/testbucket/dir/
+
+    8. Issue a dry-run heal operation to list all uploads to be healed under 'dir' prefix (WIP)
+       $ {{.HelpName}} --incomplete --recursive --dry-run play/testbucket/dir/
+
+    9. Heal all objects under 'dir' prefix with statistics-only output
+       $ {{.HelpName}} --recursive --statistics-only play/testbucket/dir/
+
+`,
+}
+
+func checkAdminHealSyntax(ctx *cli.Context) {
+	if len(ctx.Args()) != 1 {
+		cli.ShowCommandHelpAndExit(ctx, "start", 1) // last argument is exit code
+	}
+}
+
+// mainAdminHealStart - the entry function of heal command
+func mainAdminHealStart(ctx *cli.Context) error {
+
+	// Check for command syntax
+	checkAdminHealSyntax(ctx)
+
+	// Get the alias parameter from cli
+	args := ctx.Args()
+	aliasedURL := args.Get(0)
+
+	isRecursive := ctx.Bool("recursive")
+	isIncomplete := ctx.Bool("incomplete")
+	isDryRun := ctx.Bool("dry-run")
+	isRemoveBadFiles := ctx.Bool("remove-bad-files")
+	isStatsOnly := ctx.Bool("statistics-only")
+
+	console.SetColor("Heal", color.New(color.FgGreen, color.Bold))
+
+	// Create a new Minio Admin Client
+	client, err := newAdminClient(aliasedURL)
+	if err != nil {
+		fatalIf(err.Trace(aliasedURL), "Cannot initialize admin client.")
+		return nil
+	}
+
+	// Compute bucket and object from the aliased URL
+	aliasedURL = filepath.ToSlash(aliasedURL)
+	splits := splitStr(aliasedURL, "/", 3)
+	bucket, prefix := splits[1], splits[2]
+
+	healPath, myerr := client.HealStart(bucket, prefix, isRecursive,
+		isDryRun, isIncomplete, isRemoveBadFiles, isStatsOnly)
+	if myerr != nil {
+		fmt.Println("Err:", myerr)
+		return nil
+	}
+
+	fmt.Printf("Heal task started on path: %s\n", healPath)
+	fmt.Println("Following heal status...")
+
+	myerr = displayAndFollowHealStatus(client, bucket, prefix)
+	if myerr != nil {
+		fmt.Println("Err:", myerr)
```

Comment:
This error style is not needed we can simply use
```go
errorIf(displayAndFollowHealStatus(client, bucket, prefix), "Unable to display follow heal status")
return nil
```

Reply:
Fixed.